### DISCUSSION TITLE
Part-DB 1.9.0 - Update existing parts from info providers, merge parts, part associations and more

### DISCUSSION THREAD

**Discussion by jbtronics** [Part-DB/Part-DB-server#441] [Announcements]
# Part-DB 1.9.0

**If you like Part-DB, please consider donating to help further development. See the sponsor button on the [homepage](https://github.com/Part-DB/Part-DB-server).**

## Update existing parts
Using the blue button in the tools tab on a part info page, you can update part info from an info provider. This means you can pull datasheets, price infos and more from digikey, octopart and more for existing parts.
You still needs to review and manually choose what infos you wanna keep

## Merge parts
You can now merge two parts together: If you have created two versions of the same part by accident, then you can now merge the two versions of the part into a single one. All information like datasheets, partLots, etc. are combined.

## Blue dot on newly created collection items
Newly created items of a collection, which were not persisted to database yet, are marked with a blue dot on the delete item: 
<img width="74" alt="image" src="https://github.com/Part-DB/Part-DB-server/assets/5410681/9b50988b-5552-45c5-adc0-55ce68aaf15b">

This is especially useful when merging parts together or update parts from info providers. 

## Part Associations
You can now associate part with each other. This can for example be used to denote alternative/compatible parts, or what part replaces an obsolete part.

# Full changelog

_Database schema changed: You need to run `php bin/console doctrine:migrations:migrate` after upgrade!_

## New features
* Added possibility to merge two parts into one
* Allow to update existing parts from info providers
* Added possibility to associate parts with each other
* Info providers can report a price related quantity for prices (PR #421 by @d-buchmann)
* Added IPN, stocked amount and storage location columns to project BOM table view (#418)
* Allow to put barcode versions of the IPN on labels (#373)
* Allow to scan IPN barcodes with the barcode scanner and added possibility to reuse existing barcodes on part lots (vendor barcodes)
* Part-DB can now automatically detect the path for the frontend dependencies, when Part-DB is not put at the webroot. No need to recompile the assets anymore (see #426)
* The search settings in the navbar gets persisted in the browser localStorage now (#429)
* Show a notice about the dependency of permissions, and show a flash warning if permissions were corrected (#435)
* Allow to specify a timestamp when a part stock operation was actually performed (#416)
* Added an option, to immediately delete a part lot if it becomes empty during a withdrawal/move operation

## Bugfixes
* Fixed behavior of BOM entries when cloning projects (PR #433 by @d-buchmann)
* Fixed database exception in certain cases (#434)
* Added an env option to allow SAML behind a reverse proxy (#436)

## Miscellaneous
* Updated dependencies
* Fixed some deprecations
* Updated documentation (PR #425 by @Pyromane)

## New Contributors
* @d-buchmann made their first contribution in https://github.com/Part-DB/Part-DB-server/pull/421

**Full Changelog**: https://github.com/Part-DB/Part-DB-server/compare/v1.8.2...v1.9.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Part-DB/Part-DB-server/releases/tag/v1.9.0'>Part-DB 1.9.0</a>.</em>

**Comment by w0hc** [comment]
Upgraded from 1.8 to 1.9.1 and run the migration command. Now I get 500 error. I can't make heads or tails of the log, there's a LOT of junk in there for each time I attempt to load the page. Mainly some permission denied messages.

**Reply by jbtronics** [reply]
have you cleared the cache afterwards?

**Reply by w0hc** [reply]
I did not. I rolled back to 1.8.2, then upgraded to 1.9.0 which seemed to work. Last I bumped to 1.9.1 and things are working.

**Reply by w0hc** [reply]
I spoke too soon. Once I tried to save any changes the 500 error came back. Clearing the cache has not helped. And I've learned that my automatic database backups aren't working, so I can't roll back to the 1.8.2 database.
[error.txt](https://github.com/Part-DB/Part-DB-server/files/13527804/error.txt)

**Reply by jbtronics** [reply]
You have run one of the commands as the wrong user, and now the permissions of Part-DBs cache folder are wrong. You must run the commands as the www-data user and not as root.

You should be able to fix this, by removing the `var/cache` folder inside the Part-DB folder in the container completely. Afterwards just try to load Part-DB in a browser and the cache will be recreated with the right permissions.
